src/main/scala/rocket/AMOALU.scala

```diff
@@ -10,6 +10,7 @@ import org.chipsalliance.cde.config.Parameters
 class StoreGen(typ: UInt, addr: UInt, dat: UInt, maxSize: Int) {
   val size = Wire(UInt(log2Up(log2Up(maxSize)+1).W))
   size := typ
+  val dat_padded = dat.pad(maxSize*8)
```

Comment:
`maxSize` is the width of the memory word. In configurations with wide DCache, maxSize > dat, since dat is xLen.

This is needed to handle that case.